File: chain-sdk/javascript-closure-angular/API/Client/UniswapV2SwapV2DTO.js
goog.provide('API.Client.UniswapV2.SwapV2DTO');

/**
 * Swap are created for each token swap within a pair.
 * @record
 */
API.Client.UniswapV2SwapV2DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.blockNumber;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.vid;

/**
 * Transaction hash plus index in Transaction swap array.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.id;

/**
 * Reference to transaction swap was included in.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.transaction;

/**
 * Timestamp of swap, used for sorted lookups.
 * @type {!Date}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.timestamp;

/**
 * Reference to pair.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.pair;

/**
 * Address that initiated the swap.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.sender;

/**
 * The EOA (Externally Owned Account) that initiated the transaction.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.from;

/**
 * Amount of token0 sold.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.amount0In;

/**
 * Amount of token1 sold.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.amount1In;

/**
 * Amount of token0 received.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.amount0Out;

/**
 * Amount of token1 received.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.amount1Out;

/**
 * Recipient of output tokens.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.to;

/**
 * @type {!API.Client.Numerics.BigInteger}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.logIndex;

/**
 * Derived amount of tokens sold in USD.
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.amountUsd;

/**
 * @type {!number}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.evaluatedPrice;

/**
 * @type {!number}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.evaluatedAmount;

/**
 * @type {!API.Client.Transactions.ETradeAggressiveSide}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.evaluatedAggressor;

/**
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.poolId;

/**
 * @type {!string}
 * @export
 */
API.Client.UniswapV2SwapV2DTO.prototype.transactionId;

File: chain-sdk/clojure/src/on_chain_api/specs/uniswap_v2_swap_v2_dto.clj
(ns on-chain-api.specs.uniswap-v2-swap-v2-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [on-chain-api.specs.numerics-big-integer :refer :all]
            [on-chain-api.specs.transactions-e-trade-aggressive-side :refer :all]
            )
  (:import (java.io File)))


(def uniswap-v2-swap-v2-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :vid) int?
   (ds/opt :id) string?
   (ds/opt :transaction) string?
   (ds/opt :timestamp) inst?
   (ds/opt :pair) string?
   (ds/opt :sender) string?
   (ds/opt :from) string?
   (ds/opt :amount_0_in) string?
   (ds/opt :amount_1_in) string?
   (ds/opt :amount_0_out) string?
   (ds/opt :amount_1_out) string?
   (ds/opt :to) string?
   (ds/opt :log_index) numerics-big-integer-spec
   (ds/opt :amount_usd) string?
   (ds/opt :evaluated_price) float?
   (ds/opt :evaluated_amount) float?
   (ds/opt :evaluated_aggressor) transactions-e-trade-aggressive-side-spec
   (ds/opt :pool_id) string?
   (ds/opt :transaction_id) string?
   })

(def uniswap-v2-swap-v2-dto-spec
  (ds/spec
    {:name ::uniswap-v2-swap-v2-dto
     :spec uniswap-v2-swap-v2-dto-data}))
